### ISSUE TITLE
add more methods for the Pop-up menu

### ISSUE THREAD

**Issue by ys928** [fltk/fltk#675]
I want to make the control's style to become more beautiful.

but when i use a class to subclass the Fl_Menu_ ,I can't find a methods to change the pop-up menu's style

for example,I can't unstyle it's box.

![image](https://user-images.githubusercontent.com/80371119/217558350-268c1b92-586f-4bd2-9b84-a919b333b272.png)
 

**I am using FLTK**
 - Version 1.4.0

I only find the follow code is for show the Pop-up menu：
```cpp
 v = menu()->pulldown(x(), y(), w(), h(), v, this, 0, 1);
```
But I don't know how I should change it.

**Comment by Albrecht-S** [comment]
Hmm, I'm not sure if this is a request to add more methods or a question about how to use popup menus (an FLTK user question). Unfortunately I can't give you an answer to your question.

I suggest to ask in our "forum" fltk.general (see below for advice). This can be used to discuss what you need or whether there's an existing solution to your problem.

Standard reply follows:

> This is the wrong place to ask questions about building or running programs with FLTK (user support). FLTK "issues" are reserved for bug reports and requests for enhancements, for further information please visit https://www.fltk.org/bugs.php.
>
>Please ask user questions in our user group [fltk.general](https://groups.google.com/forum/#!forum/fltkgeneral). Your question will have a wider audience and you will likely get more and quicker replies.
>
> Please include the system/platform (Windows, Linux, macOS), the build system (Visual Studio, CMake, Ninja), compiler (gcc, clang), compiler version, and build error messages in text form (copy and paste preferred). TIA.

**Comment by MatthiasWM** [comment]
If you use an Fl_Menu_Bar, the popup menu uses the color and box style from that menubar. If you used the popup() or pulldown() call, you can add a reference to an Fl_Menu_ item, and the popup will also copy that style. If the Fl_Menu_ has no parent, you will get the popup behaviour, otherwise, you get the pulldown behaviour.

**Comment by ys928** [comment]
emm,I think it may be a bug or design error .

for example the follow simple  code:
```cpp
Fl_Menu_Item menuBar[] = {
    {"File",0,0,0,FL_SUBMENU},
        {"Open"},
        {0},
    {"Edit"},
    {"Setting"},
    {0}
};

int main(int argc, char** argv) {
	Fl_Window win(100, 100, 800, 600, "test");
	{
          Fl_Menu_Bar* menu = new Fl_Menu_Bar(0, 100, 800, 30, "menu");
          menu->box(FL_NO_BOX);
          menu->down_box(FL_NO_BOX);
          menu->menu(menuBar);
	}
	win.end();
    win.show();
    Fl::set_color(FL_FOREGROUND_COLOR, 0xD3D3D300);
    Fl::set_color(FL_BACKGROUND_COLOR, 0x1e1e1e00);
	Fl::run();
}
```

but the subMenu still have the box style

![image](https://user-images.githubusercontent.com/80371119/217834566-e3868d60-4bdd-413e-ad94-e0415e86971a.png)

I hava setted box()  and  down_box() to  FL_NO_BOX

**Comment by MatthiasWM** [comment]
Sigh, yes, that is unexpected and I would categorise it as a bug. In line 375 of Fl_Menu.cxx, it disallows FL_NO_BOX, which is fine (you should not use that one), but FL_FLAT_BOX (which would be the right type to use) is also changed into FL_UP_BOX.

You can create your own box type based on the flat box by calling `Fl::set_boxtype(FL_FREE_BOXTYPE, FL_FLAT_BOX);` once and then setting your menubar to `box(FL_FREE_BOXTYPE);`.

**Comment by Albrecht-S** [comment]
@ys928 Thanks for posting the code, this is always helpful.

Note: I reclassified this as a bug report (as @MatthiasWM wrote above).

**Comment by Albrecht-S** [comment]
Reopened, since this is now classified as a bug report.

@MatthiasWM Do you think you can fix this, is it possible to fix it at all?

**Comment by MatthiasWM** [comment]
Yes, I will remove the code that overrides FL_FLAT_BOX to FL_UP_BOX. There is no reason for that. Not sure yet what to do about FL_NO_BOX. Maybe it should force FL_FLAT_BOX as that seems to be the intention. No box would draw no background with all its known issues.

**Comment by MatthiasWM** [comment]
Fixed in 72f860438170638d6aa492b477a59ff88b565d9d
